newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/context/InstrumentationContextManager.java

```diff
@@ -69,7 +68,7 @@ public InstrumentationContextManager(Instrumentation instrumentation) {
         this.classWeaverService = new ClassWeaverService(instrumentation);
 
         // these matchers only modify the InstrumentationContext, they don't actually transform classes
-        matchVisitors.put(new ScalaTraitMatcher(), new ScalaTraitFinalFieldTransformer());
```

Comment:
ScalaTraitFinalFieldTransformer moved from matchVisitors as it was marking all scala classes with final fields as modified for instrumentation.